### PR TITLE
RavenDB-14234: TrySimplifyJson should assign the right type to derive…

### PR DESCRIPTION
**Pull request by garayx**
…d properties

### CODE REVIEW

File: src/Raven.Client/Documents/Session/EntityToBlittable.cs [hunk 1]
```diff
@@ -452,12 +452,34 @@ private static bool ShouldSimplifyJsonBasedOnType(string typeValue)
...
+            }
+            catch (AmbiguousMatchException)
+            {
+                var memberInfos = ReflectionUtil.GetPropertiesAndFieldsFor(rootType, BindingFlags.Public | BindingFlags.Instance | BindingFlags.NonPublic)
+                    .Where(x => x.Name == propertyName)
+                    .ToList();
+
+                while (typeof(object) != rootType)
+                {
+                    memberInfo = memberInfos.FirstOrDefault(x => x.DeclaringType == rootType);
```
Comment (ayende) [inline src/Raven.Client/Documents/Session/EntityToBlittable.cs:473]: IsAssignableFrom, instead?

Comment (ppekrol) [reply src/Raven.Client/Documents/Session/EntityToBlittable.cs:473]: we want to get 'closest' decendant and IsAssignableFrom returns any, or I'm wrong?

Comment (ayende) [reply src/Raven.Client/Documents/Session/EntityToBlittable.cs:473]: But you are missing interfaces

Comment (garayx) [reply src/Raven.Client/Documents/Session/EntityToBlittable.cs:473]: Added test with classes that inherit from interface

**Review by ppekrol — APPROVED** [review]

**Review by ayende — APPROVED** [review]

**Pull request merged**